modules/redis/scanner.go

```diff
@@ -228,44 +362,97 @@ func (scanner *Scanner) Scan(target zgrab2.ScanTarget) (zgrab2.ScanStatus, inter
 	}
 	defer scan.Close()
 	result := scan.result
-	pingResponse, err := scan.SendCommand("PING")
+	pingResponse, err := scan.SendCommand(scanner.commandMappings["PING"])
 	if err != nil {
-		// if the first command fails (as opposed to succeeding but returning an
+		// If the first command fails (as opposed to succeeding but returning an
 		// ErrorMessage response), then flag the probe as having failed.
 		return zgrab2.TryGetScanStatus(err), nil, err
 	}
 	// From this point forward, we always return a non-nil result, implying that
 	// we have positively identified that a redis service is present.
 	result.PingResponse = forceToString(pingResponse)
 	if scanner.config.Password != "" {
-		authResponse, err := scan.SendCommand(scanner.config.AuthCommand, scanner.config.Password)
+		authResponse, err := scan.SendCommand(scanner.commandMappings["AUTH"], scanner.config.Password)
 		if err != nil {
 			return zgrab2.TryGetScanStatus(err), result, err
 		}
 		result.AuthResponse = forceToString(authResponse)
 	}
-	infoResponse, err := scan.SendCommand("INFO")
+	infoResponse, err := scan.SendCommand(scanner.commandMappings["INFO"])
 	if err != nil {
 		return zgrab2.TryGetScanStatus(err), result, err
 	}
 	result.InfoResponse = forceToString(infoResponse)
-	infoResponseBulk, ok := infoResponse.(BulkString)
-	if ok {
+	if infoResponseBulk, ok := infoResponse.(BulkString); ok {
+		fieldsWeAreLookingFor := 12
+		fieldsFound := 0
 		for _, line := range strings.Split(string(infoResponseBulk), "\r\n") {
-			if strings.HasPrefix(line, "redis_version:") {
-				result.Version = strings.SplitN(line, ":", 2)[1]
+			switch line_prefix_suffix := strings.SplitN(line, ":", 2); line_prefix_suffix[0] {
+			case "redis_version":
+				fieldsFound += 1
+				result.Version = line_prefix_suffix[1]
+			case "os":
+				fieldsFound += 1
+				result.OS = line_prefix_suffix[1]
+			case "arch_bits":
+				fieldsFound += 1
+				result.ArchBits = line_prefix_suffix[1]
+			case "redis_mode":
+				fieldsFound += 1
+				result.Mode = line_prefix_suffix[1]
+			case "redis_git_sha1":
+				fieldsFound += 1
+				result.GitSha1 = line_prefix_suffix[1]
+			case "redis_build_id":
+				fieldsFound += 1
+				result.BuildID = line_prefix_suffix[1]
+			case "gcc_version":
+				fieldsFound += 1
+				result.GCCVersion = line_prefix_suffix[1]
+			case "mem_allocator":
+				fieldsFound += 1
+				result.MemAllocator = line_prefix_suffix[1]
+			case "uptime_in_seconds":
+				fieldsFound += 1
+				result.Uptime = convToUint32(line_prefix_suffix[1])
+			case "used_memory":
+				fieldsFound += 1
+				result.UsedMemory = convToUint32(line_prefix_suffix[1])
+			case "total_connections_received":
+				fieldsFound += 1
+				result.ConnectionsReceived = convToUint32(line_prefix_suffix[1])
+			case "total_commands_processed":
+				fieldsFound += 1
+				result.CommandsProcessed = convToUint32(line_prefix_suffix[1])
+			}
+			if fieldsWeAreLookingFor == fieldsFound {
```

Comment:
Using a flat counter means that duplicates will be treated the same as fields you are still looking for (to take an obvious case, say there are two dozen `os:foo\r\n` followed by the rest of the expected fields)

Reply:
Yeah, I was thinking about that. The two alternatives I thought of was either to add a ton of bools and at the end have a large if statement, or to remove the final check to break early. Which solution do you think would be best?

Reply:
I'd just ditch the break -- it's already walking the string once to do the split, so walking the whole resulting array shouldn't noticeably hurt the performance (unless it's already bad, in which case the split itself would be the problem, not walking the resulting array).